lib/shelley/test/integration/Main.hs

```diff
@@ -220,25 +226,42 @@ specWithServer (tr, tracers) = aroundAll withContext . after tearDown
         race (takeMVar ctx >>= action') (withServer setupContext) >>=
             either pure (throwIO . ProcessHasExited "integration")
 
-    withServer onStart = bracketTracer' tr "withServer" $ do
+    withServer action = bracketTracer' tr "withServer" $ do
         minSev <- nodeMinSeverityFromEnv
-        let tr' = contramap MsgCluster tr
+        testPoolConfigs' <- poolConfigsFromEnv
         withSystemTempDir tr' "test" $ \dir ->
-            withCluster tr' minSev testPoolConfigs dir $
-                \socketPath block0 (gp, vData) ->
-                    withTempDir tr' dir "wallets" $ \db -> do
-                        serveWallet @(IO Shelley)
-                            (SomeNetworkDiscriminant $ Proxy @'Mainnet)
-                            tracers
-                            (SyncTolerance 10)
-                            (Just db)
-                            "127.0.0.1"
-                            ListenOnRandomPort
-                            Nothing
-                            socketPath
-                            block0
-                            (gp, vData)
-                            (onStart gp)
+            withCluster
+                tr'
+                minSev
+                testPoolConfigs'
+                dir
+                onByron
+                (afterFork dir)
+                (onClusterStart action dir)
+
+    tr' = contramap MsgCluster tr
+    onByron _ = pure ()
+    afterFork dir _ = do
+        let encodeAddr = T.unpack . encodeAddress @'Mainnet
+        let addresses = map (first encodeAddr) shelleyIntegrationTestFunds
+        sendFaucetFundsTo stdoutTextTracer dir addresses
```

Comment:
The setup is sprinkled across integration/Main.hs, Faucet.hs, and Launch.hs, which feels a bit weird. Maybe we can resolve that sometime.